modules/administration-guide/partials/proc_migration-to-devworkspace-engine.adoc

```diff
@@ -0,0 +1,208 @@
+
+include::partial$snip_creating-che-database.adoc[]
+
+. Migrate {prod-short} users:
++
+[subs="+quotes,+attributes"]
+----
+DUMP="che.sql"
+ALL_USERS={prod-id}-users.txt
+while IFS= read -r line
+do
+  IDS=($line)
+  USER_ID=${IDS[0]}
+  OPENSHIFT_USER_ID=${IDS[1]}
+
+  sed -i -e "s|${USER_ID}|${OPENSHIFT_USER_ID}|g" "${DUMP}"
+
+  echo "[INFO] Migrated User ID from \"${USER_ID}\" to \"${OPENSHIFT_USER_ID}\""
+done < "${ALL_USERS}"
+----
+
+include::partial$snip_restoring-che-database.adoc[]
+
+. Add empty users profiles:
++
+[subs="+quotes,+attributes"]
+----
+ALL_USERS={prod-id}-users.txt
+while IFS= read -r line
+do
+  IDS=($line)
+  OPENSHIFT_USER_ID=${IDS[1]}
+
+  {orch-cli} exec $POSTGRES_POD -n {prod-namespace}  -- bash  -c "psql ${CHE_POSTGRES_DB} -tAc \"insert into profile(userid) values ('${OPENSHIFT_USER_ID}');\""
+
+  echo "[INFO] Added profile for \"${OPENSHIFT_USER_ID}\""
+done < "${ALL_USERS}"
+----
+
+. Find Cluster Service Version name:
++
+[subs="+quotes,+attributes"]
+----
+CSV=$({orch-cli} get subscription {prod-id} -n {prod-namespace} -o jsonpath="{.status.currentCSV}")
+----
+
+. Delete Cluster Service Version:
++
+[subs="+quotes,+attributes"]
+----
+{orch-cli} delete csv ${CSV} -n {prod-namespace}
+----
+
+. Delete Subscription:
++
+[subs="+quotes,+attributes"]
+----
+{orch-cli} delete subscription {prod-id} -n {prod-namespace}
+----
+
+. Enable {devworkspace} engine:
++
+[subs="+quotes,+attributes"]
+----
+{orch-cli} patch checluster/{prod-checluster} -n {prod-namespace} --type=json -p \
+'[{"op": "replace", "path": "/spec/devWorkspace/enable", "value": true}]'
+----
+
+. Enable single-host exposure strategy:
++
+[subs="+quotes,+attributes"]
+----
+{orch-cli} patch checluster/{prod-checluster} -n {prod-namespace} --type=json -p \
+'[{"op": "replace", "path": "/spec/server/serverExposureStrategy", "value": "single-host"}]'
+----
+
+. Delete {identity-provider} route:
++
+[subs="+quotes,+attributes"]
+----
+{orch-cli} delete route keycloak -n {prod-namespace}
+----
+
+. Delete {identity-provider} service:
++
+[subs="+quotes,+attributes"]
+----
+{orch-cli} delete service keycloak -n {prod-namespace}
+----
+
+. Delete {identity-provider} deployment:
++
+[subs="+quotes,+attributes"]
+----
+{orch-cli} delete deployment keycloak -n {prod-namespace}
+----
+
+. Create a new subscription to `{prod-tech-preview-channel}` channel:
++
+[subs="+quotes,+attributes"]
+----
+{orch-cli} apply -f - <<EOF
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: {prod-id}
+  namespace: openshift-operators
+spec:
+  channel: {prod-tech-preview-channel}
+  installPlanApproval: Automatic
+  name: {prod-tech-preview-olm-package}
+  source: {prod-catalog-source}
+  sourceNamespace: openshift-marketplace
+EOF
+----
+
+. Wait until Operator is ready:
++
+[subs="+quotes,+attributes"]
+----
+{orch-cli} wait --for=condition=ready pod -l app.kubernetes.io/component={prod-operator} -n openshift-operators --timeout=120s
```

Comment:
I was following the steps, but when I copy-pasted and ran the command it printed:
```
error: no matching resources found
```
And only after 10 or 15 seconds of delay it started to work and wait (I was trying to rerun it during that time). The reason is that the pod hadn't been created when I tried to run the command. We should handle such situation, because it might be confusing for users not familiar with K8s internals.

Reply:
Ok